pkg/url/url_test.go

```diff
@@ -1637,3 +1638,388 @@ func TestGetContainerURL(t *testing.T) {
+	err := esi.SetConfiguration("url", testURL2)
+	if err != nil {
+		// discard the error, since no physical file to write
+		t.Log("Expected error since no physical env file to write")
+	}
+	err = esi.SetConfiguration("url", testURL3)
+	if err != nil {
+		// discard the error, since no physical file to write
+		t.Log("Expected error since no physical env file to write")
+	}
+	fkclient, fkclientset := kclient.FakeNew()
+	fkclient.Namespace = "default"
+	fkclientset.Kubernetes.PrependReactor("list", "ingresses", func(action ktesting.Action) (bool, runtime.Object, error) {
+		ingress := extensionsv1.IngressList{
+			Items: pushedIngress,
+		}
+		return true, &ingress, nil
+	})
+
+	tests := []struct {
+		name      string
+		component string
+		client    *kclient.Client
+		wantURLs  []URL
+	}{
+		{
+			name:      "Should retrieve the URL list",
+			component: componentName,
+			client:    fkclient,
+			wantURLs: []URL{
+				URL{
+					TypeMeta:   metav1.TypeMeta{Kind: "url", APIVersion: "odo.dev/v1alpha1"},
+					ObjectMeta: metav1.ObjectMeta{Name: testURL1.Name},
+					Spec:       URLSpec{Host: "ingressurl1.com", Port: testURL1.Port, Secure: testURL1.Secure},
+					Status: URLStatus{
+						State: StateTypeLocallyDeleted,
+					},
+				},
+				URL{
+					TypeMeta:   metav1.TypeMeta{Kind: "url", APIVersion: "odo.dev/v1alpha1"},
+					ObjectMeta: metav1.ObjectMeta{Name: testURL2.Name},
+					Spec:       URLSpec{Host: "ingressurl2.com", Port: testURL2.Port, Secure: testURL2.Secure},
+					Status: URLStatus{
+						State: StateTypePushed,
+					},
+				},
+				URL{
+					TypeMeta:   metav1.TypeMeta{Kind: "url", APIVersion: "odo.dev/v1alpha1"},
+					ObjectMeta: metav1.ObjectMeta{Name: testURL3.Name},
+					Spec:       URLSpec{Host: "ingressurl3.com", Port: testURL3.Port, Secure: testURL3.Secure, TLSSecret: componentName + "-tlssecret"},
+					Status: URLStatus{
+						State: StateTypeNotPushed,
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			urls, err := ListIngressURL(tt.client, esi, componentName)
+			if err != nil {
+				t.Errorf("unexpected error %v", err)
+			}
+
+			if len(urls.Items) != len(tt.wantURLs) {
+				t.Errorf("numbers of url listed does not match, expected %v, got %v", len(tt.wantURLs), len(urls.Items))
+			}
+			actualURLMap := make(map[string]URL)
+			for _, actualURL := range urls.Items {
+				actualURLMap[actualURL.Name] = actualURL
+			}
+			for _, wantURL := range tt.wantURLs {
+				if !reflect.DeepEqual(actualURLMap[wantURL.Name], wantURL) {
+					t.Errorf("Expected %v, got %v", wantURL, actualURLMap[wantURL.Name])
+				}
+			}
+		})
+	}
+
+}
+
+func TestGetIngress(t *testing.T) {
+	// initialising the fakeclient
+	componentName := "testcomponent"
+	componentLabel := componentlabels.ComponentLabel
+	urlNameLabel := "odo.openshift.io/url-name"
+	port := intstr.IntOrString{
+		Type:   intstr.Int,
+		IntVal: int32(8080),
+	}
+	ingressURL1 := extensionsv1.Ingress{
```

Comment:
you can use the kclient fake ingresses here

Reply:
updated